pkg/gather/aws/aws.go

```diff
@@ -0,0 +1,236 @@
+	// session will be created based on the usual credential configuration
+	// (AWS_PROFILE, AWS_ACCESS_KEY_ID, etc.).
+	session *session.Session
+}
+
+// New returns an AWS Gather from ClusterMetadata.
+func New(logger logrus.FieldLogger, serialLogBundle string, bootstrap string, masters []string, metadata *types.ClusterMetadata) (providers.Gather, error) {
+	filters := make([]Filter, 0, len(metadata.ClusterPlatformMetadata.AWS.Identifier))
+	for _, filter := range metadata.ClusterPlatformMetadata.AWS.Identifier {
+		filters = append(filters, filter)
+	}
+	region := metadata.ClusterPlatformMetadata.AWS.Region
+	session, err := awssession.GetSessionWithOptions(
+		awssession.WithRegion(region),
+		awssession.WithServiceEndpoints(region, metadata.ClusterPlatformMetadata.AWS.ServiceEndpoints),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Gather{
+		logger:          logger,
+		region:          region,
+		filters:         filters,
+		session:         session,
+		serialLogBundle: serialLogBundle,
+		bootstrap:       bootstrap,
+		masters:         masters,
+		directory:       filepath.Dir(serialLogBundle),
+	}, nil
+}
+
+// Run is the entrypoint to start the gather process.
+func (g *Gather) Run() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	defer cancel()
+
+	awsSession := g.session
+	if awsSession == nil {
+		var err error
+		// Relying on appropriate AWS ENV vars (eg AWS_PROFILE, AWS_ACCESS_KEY_ID, etc)
+		awsSession, err = session.NewSession(aws.NewConfig().WithRegion(g.region))
+		if err != nil {
+			return err
+		}
+	}
+	awsSession.Handlers.Build.PushBackNamed(request.NamedHandler{
+		Name: "openshiftInstaller.OpenshiftInstallerUserAgentHandler",
+		Fn:   request.MakeAddToUserAgentHandler("OpenShift/4.x Gather", version.Raw),
+	})
+
+	ec2Client := ec2.New(awsSession)
+
+	instances, err := g.findEC2Instances(ctx, ec2Client)
+	if err != nil {
+		return err
+	}
+
+	if len(instances) == 0 {
+		g.logger.Infoln("Skipping console log gathering: no instances found")
+		return nil
+	}
+
+	serialLogBundleDir := strings.TrimSuffix(filepath.Base(g.serialLogBundle), ".tar.gz")
+	filePathDir := filepath.Join(g.directory, serialLogBundleDir)
+	err = os.MkdirAll(filePathDir, 0755)
+	if err != nil && !errors.Is(err, os.ErrExist) {
+		return err
+	}
+
+	var errs []error
+	var files []string
+	for _, instance := range instances {
+		filePath, err := g.downloadConsoleOutput(ctx, ec2Client, instance, filePathDir)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			files = append(files, filePath)
+		}
+	}
+
+	if len(files) > 0 {
+		err := gather.CreateArchive(files, g.serialLogBundle)
+		if err != nil {
+			errs = append(errs, errors.Wrap(err, "failed to create archive"))
+		}
+	}
+
+	if err := gather.DeleteArchiveDirectory(filePathDir); err != nil {
+		// Note: cleanup is best effort, it shouldn't fail the gather
+		g.logger.Debugf("failed to remove archive directory: %v", err)
```

Comment:
nit: unlike error messages, log messages are capitalized
```suggestion
		g.logger.Debugf("Failed to remove archive directory: %v", err)
```